ratelimit/core.py

```diff
@@ -59,19 +59,19 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
                     user, group = await self.authenticate(scope)
                 except Exception as exc:
                     if self.on_auth_error is not None:
-                        reponse = await self.on_auth_error(exc)
-                        return await reponse(scope, receive, send)
+                        response = await self.on_auth_error(exc)
+                        return await response(scope, receive, send)
                     raise exc
 
                 # Select the first rule that can be matched
-                _rules = [rule for rule in rules if group == rule.group]
-                if _rules:
-                    rule = _rules[0]
+                rule = next(filter(lambda r: r.group == group, rules), None)
+                if rule:
                     break
         else:  # If no rule can match, run `self.app` and return
             return await self.app(scope, receive, send)
 
-        if not [name for name in RULENAMES if getattr(rule, name) is not None]:
+        # Find if rule have any of required attributes
+        if not any(getattr(rule, name) for name in RULENAMES):
```

Comment:
Should use `any(getattr(rule, name) is not None for name in RULENAMES)`

Reply:
OK , I will check it